tests/PHPStan/Analyser/data/preg_match_shapes.php

```diff
@@ -0,0 +1,161 @@
+<?php
+
+namespace PregMatchShapes;
+
+use function PHPStan\Testing\assertType;
+
+
+function doMatch(string $s): void {
+	if (preg_match('/Price: /i', $s, $matches)) {
+		assertType('array{string}', $matches);
+	}
+	assertType('array<string>', $matches);
```

Comment:
```suggestion
	} else {
		assertType('array{}', $matches);
	}
	assertType('array{string}', $matches);
```

when pattern is not matched, the `$matches` is reset to `array{}` - https://3v4l.org/NE0bm

Reply:
fixed